components/brave_ads/core/internal/user_engagement/conversions/resource/conversion_resource.h

```diff
@@ -34,7 +32,9 @@ class ConversionResource final : public AdsClientNotifierObserver {
     return manifest_version_;
   }
 
-  const std::optional<ConversionResourceInfo>& get() const { return resource_; }
+  base::optional_ref<const ConversionResourceInfo> get() const {
```

Comment:
For cases where we are returning a reference to an optional member we should use `optional<T>&` as that is safer, as we don't have any risk of creating copies, etc.

Reply:
To clarify, we shouldn't return `base::optional_ref` for cases a `const optional<T>&` is what we are accessing, because we could end up with a dangling reference.

Reply:
Thanks, fixing now

Reply:
Fixed, and pushed